### PR TITLE
add emergency chat room

### LINKED ISSUE #53: EmergencyChatPage

### PR DESCRIPTION
**Pull request by DasProffi**
closes #53

### CODE REVIEW

**Comment by DasProffi** [comment]
| Light | Dark |
| -- | -- |
| |  |

File: lib/pages/emergency_chat.dart [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                      child: Center(
+                        child: Ink(
+                          decoration: ShapeDecoration(
+                            color: Theme.of(context).colorScheme.secondary,
+                            shape: const CircleBorder(),
+                          ),
+                          child: IconButton(
+                            icon: Icon(
+                              Icons.send,
+                              size: iconSizeSmall,
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:113]: make it even smaller

File: lib/pages/emergency_chat.dart [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                      messages.map((e) => ChatMessage(message: e)).toList(),
+                ),
+              ),
+              Container(
+                width: mediaQuery.width,
+                padding: const EdgeInsets.all(paddingSmall),
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: TextField(
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:80]: add placeholder text

File: lib/pages/emergency_chat.dart [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    const OutlineInputBorder outlineInputBorder = OutlineInputBorder(
+      borderRadius: BorderRadius.all(
+        Radius.circular(borderRadiusMedium),
+      ),
+    );
+
+    return Scaffold(
+      appBar: AppBar(
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:38]: An AppBar shadow would be nice

File: lib/pages/emergency_chat.dart [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    {"isFromEmergencyRoom": true, "message": "okay..."}
+  ];
+  final TextEditingController _controller = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    Size mediaQuery = MediaQuery.of(context).size;
+
+    addMessage() {
+      messages.add({"isFromEmergencyRoom": false, "message": _controller.text});
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:27]: `setState`

Comment (use-to) [inline lib/pages/emergency_chat.dart:27]: trim the message

File: lib/pages/emergency_chat.dart [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                    ),
+                    Material(
+                      color: Colors.transparent,
+                      child: Center(
+                        child: Ink(
+                          decoration: ShapeDecoration(
+                            color: Theme.of(context).colorScheme.secondary,
+                            shape: const CircleBorder(),
+                          ),
+                          child: IconButton(
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:110]: disabled this button while there is no input (only spaces included)

File: lib/pages/emergency_chat.dart [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                  .copyWith(fontSize: fontSizeMedium),
+            ),
+          ],
+        ),
+        centerTitle: false,
+      ),
+      body: FutureBuilder(
+        builder: (context, snapshot) {
+          return Column(
+            children: [
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:67]: add some spacing here

Comment (use-to) [reply lib/pages/emergency_chat.dart:67]: `topMargin`

File: lib/pages/emergency_chat.dart [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        Radius.circular(borderRadiusMedium),
+      ),
+    );
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            const CircleAvatar(
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:42]: remove this

**Review by use-to — CHANGES_REQUESTED** [review]

File: lib/pages/emergency_chat.dart [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                                canSend = false;
+                              })
+                            }
+                          else if (_controller.text.trim().isNotEmpty &&
+                              !canSend)
+                            {
+                              setState(() {
+                                canSend = true;
+                              })
+                            }
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:131]:
```suggestion
                          bool couldSend = _controller.text.trim().isNotEmpty;
                          if (canSend != couldSend) {
                            setState(() {
                              canSend = couldSend;
                            });
                          }
```

File: lib/pages/emergency_chat.dart [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+                      width: distanceSmall,
+                    ),
+                    Material(
+                      color: Colors.transparent,
+                      child: Center(
+                        child: Ink(
+                          decoration: ShapeDecoration(
+                            color: canSend
+                                ? positiveColor
+                                : Theme.of(context).colorScheme.secondary,
```
Comment (use-to) [inline lib/pages/emergency_chat.dart:145]: Is there no button `disabled` color in theme definition?

**Review by use-to — CHANGES_REQUESTED** [review]

**Review by use-to — APPROVED** [review]

**Pull request merged**